warehouse/query-core/src/test/java/datawave/query/data/parsers/EventKeyTest.java

```diff
@@ -59,6 +59,10 @@ public void testRepeatedParsing() {
 
         parser.parse(eventKeyWithChildUidNoDashes);
         assertKeyWithChildUidNoDashes(parser);
+
+        // this should skip actual processing, since it was already parsed up once.
+        parser.parse(eventKeyWithChildUidNoDashes);
```

Comment:
this does not actually test that it is not parsed again.  Perhaps you can use a separate key that equals but is not == to the original key.  Then you can test that the cached pointer still points to the original one.

Reply:
sure, will update this one the next commit